src/edu/ucsb/nceas/metacat/doi/datacite/DefaultDataCiteFactory.java

```diff
@@ -0,0 +1,119 @@
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+package edu.ucsb.nceas.metacat.doi.datacite;
+
+import java.text.SimpleDateFormat;
+
+import org.apache.log4j.Logger;
+import org.dataone.client.v2.itk.D1Client;
+import org.dataone.service.exceptions.InvalidRequest;
+import org.dataone.service.exceptions.ServiceFailure;
+import org.dataone.service.types.v1.Identifier;
+import org.dataone.service.types.v2.Node;
+import org.dataone.service.types.v2.ObjectFormat;
+import org.dataone.service.types.v2.SystemMetadata;
+import org.w3c.dom.Document;
+
+import edu.ucsb.nceas.ezid.profile.DataCiteProfile;
+import edu.ucsb.nceas.ezid.profile.DataCiteProfileResourceTypeValues;
+import edu.ucsb.nceas.ezid.profile.ErcMissingValueCode;
+import edu.ucsb.nceas.metacat.dataone.MNodeService;
+
+
+/**
+ * Default factory to generate a simple datacite metadata xml string
+ * @author tao
+ *
+ */
+public class DefaultDataCiteFactory extends DataCiteMetadataFactory {
+    private static Logger logMetacat = Logger.getLogger(DefaultDataCiteFactory.class);
+    
+    
+    /**
+     * Method to generate the data cite xml document
+     */
+    @Override
+    public String generateMetadata(Identifier identifier, SystemMetadata sysmeta) throws InvalidRequest, ServiceFailure {
+        if(identifier != null && sysmeta != null) {
+            try {
+                String language = "English";
+                Document doc = generateROOTDoc();
+                //identifier
+                String scheme = DOI;
+                String id = removeIdSchemePrefix(identifier.getValue(), scheme);
+                addIdentifier(doc, id, scheme);
+                
+                //creator
+                String affiliation = null;
+                String nameIdentifier = null;
+                String nameIdentifierSchemeURI = null;
+                String nameIdentifierScheme = null;
+                appendCreator(sysmeta.getRightsHolder().getValue(), doc, affiliation, nameIdentifier, nameIdentifierSchemeURI, nameIdentifierScheme);
+                
+                //title
+                appendTitle(ErcMissingValueCode.UNKNOWN.toString(), doc, language);
+                
+                //publish
+                Node node = MNodeService.getInstance(null).getCapabilities();
+                addPublisher(doc,node.getName());
```

Comment:
While using the Member Node name as a publisher is a good fallback, we should be consulting the EML or ISO metadata here and inserting the publisher string that is in the metadata.  If we don't have it, using the MN name seems fine.  But what happens when this Metacat installation isn't part of DataONE, and the DataONE properties aren't populated? We should be testing for non-empty values before using them.